pkg/pillar/cmd/monitor/monitor.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package monitor
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentbase"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	agentName            = "monitor"
+	errorTime            = 3 * time.Minute
+	warningTime          = 40 * time.Second
+	stillRunningInterval = 25 * time.Second
+)
+
+var logger *logrus.Logger
+var log *base.LogObject
+
+type monitorContext struct {
```

Comment:
```suggestion
type monitor struct {
```
It is not a context, is it?

Reply:
@christoph-zededa we call the field in SubscriptionOptions Ctx aka context, and the type is typically "$agent"Context.

Sure, this is not the golang context.Context. But there is more than one kind of context; which one you refer to depends on your context ;-)

Reply:
I would side with Christoph on this one, I think it is cleaner (and Go-Idiomatic) to just call it `monitor`. I haven't see this context terminology used with custom structs in any other Go project.

Reply:
@eriknordmark @milan-zededa @christoph-zededa so, which one should I use? I have no personal preferences but as Erik mentioned we use this style all over EVE

Reply:
NIM, zedrouter, mmagent and their components no longer use the Context suffix. I think it reads better and avoids confusion with `context.Context`